Added JPEGPhoto and createhomedir

gniemetz opened the pull request:
My first try to build a "real" python program, so please be patient :)

I've added the support for creating JPEGPhoto, also added support to create the home directory

Thanks you!

best regards
Gerd

gniemetz committed:
Added JPEGPhoto and createhomedir

gregneagle commented on createuserpkg:
```diff
@@ -36,16 +35,20 @@ def main():
-        '--name', '-n', help='User shortname. REQUIRED.')
+        '--name', '-n',
+        help='User shortname. REQUIRED.')
     required_user_options.add_option(
```
You've made a bunch of changes unrelated to your core changes. This makes reviewing your changes harder.

gniemetz replied:
sorry about that, thought it would be easier to read

gregneagle commented on createuserpkg:
```diff
@@ -119,7 +135,10 @@ def main():
         user_data['IsHidden'] = u'YES'
-
+    if options.picture:
+        from locallibs import imageutil
```
imports really belong at the top of a Python file unless you have a really good reason to do them conditionally.

gniemetz replied:
I thought i only import it if there is a image as argument, but i will put it on top

gregneagle commented on locallibs/imageutil.py:
```diff
@@ -0,0 +1,62 @@
+            image_path,
+            temp_image)
+        try:
+            subprocess.call(cmd, shell=True)
```
Please don't use shell=True with subprocess. Properly format your command arguments into a list and use the command list with subprocess.call.

gniemetz replied:
will do

gregneagle commented on locallibs/userpkg.py:
```diff
@@ -88,6 +89,9 @@
 
+PI_CREATEHOMEDIR = """
+/usr/sbin/createhomedir -c
+"""
```
Using createhomedir will only work when a package is installed to the current boot volume. All the other options for pycreateuserpkg work no matter if the package target is the current boot volume or an alternate volume (as when installing into an AutoDMG image, or via a Imagr or DeployStudio or Bootstrappr workflow). I'm hesitant to merge a feature that works only sometimes.

poundbangbash replied:
In the latest 10.13.4 beta 4 there is also a new behavior to createmobileaccount that prompts in the cli run for an admin account to set the SecureToken. There is a `-D` option to suppress it. There should be some logic to detect the OS and act accordingly.

-Eric

gregneagle replied:
`createmobileaccount` and `createhomedir` are different tools; I don't know if there are similar new issues in 10.13.4 that would affect `createhomedir`.

gniemetz replied:
I will put in into the if block, under the killall statements

gregneagle reviewed (commented):
See in-line comments.

gniemetz committed:
Adjusted according Greg Neagles comments

- Imports of modules on top
- Changed to subprocess call
- Put the createhomedir into the if-block

gregneagle commented on locallibs/userpkg.py:
```diff
@@ -88,6 +88,9 @@
 
+PI_CREATEHOMEDIR = """
+    /usr/sbin/createhomedir -c
+"""
```
I don't think you've addressed my issue with `createhomedir` not working for anything other than installs to /.
I think the postinstall script bit needs to test $3 and do the right thing, and I think the overall `createuserpkg` tool needs to warn people in the help output that this option works only in some scenarios; etc.

gniemetz replied:
I will add the createhomedir command to the pi_live_actions, so it gets written into the if [ "$3" == "/" ] block. Additionally i will add a notice to the help text that says that it only work an a boot volume.
Would that be ok?
Thanks again
Gerd

gniemetz committed:
Refined help message for createhomedir

gniemetz commented:
Have you had time to look at my latest pull request

Thanks in advance
regards
Gerd

gregneagle commented:
Not yet, no.

gregneagle commented:
Sorry, coming back to this now after a billion-year break; did you want to still work on merging this?
